pkg/infrastructure/services/image.go

```diff
@@ -89,22 +116,23 @@ func (i *imageHandler) resolveImageNameTag() string {
 	return fmt.Sprintf("%s:%s", name, i.resolveTag())
 }
 
-// resolves like "latest"
+// resolves like "latest", 0.8.0, and so on
 func (i *imageHandler) resolveTag() string {
 	if len(i.image.Tag) == 0 {
-		return defaultImageTag
+		return version.Version
 	}
 	return i.image.Tag
 }
 
-func newImageHandler(instance v1alpha1.KogitoService, imageName string, cli *client.Client) *imageHandler {
+func newImageHandler(instance v1alpha1.KogitoService, defaultImageName string, cli *client.Client) *imageHandler {
 	if instance.GetSpec().GetImage() == nil {
 		instance.GetSpec().SetImage(v1alpha1.Image{})
 	}
 	handler := &imageHandler{
 		image:            instance.GetSpec().GetImage(),
 		imageStream:      nil,
-		defaultImageName: imageName,
+		defaultImageName: defaultImageName,
+		client:           cli,
 	}
 
 	if !cli.IsOpenshift() {
```

Comment:
return handler at the end and if Openshift, execute more code on handler
It would make more sense to me, like you do in `resolveImage()`